### PR TITLE
[fix] Raise a fatal error in the ROM if max. limit of measurements is crossed

### LINKED ISSUE #1590: Error for Stash Measurement MB Command Causes Fatal Error
Within Caliptra, if the SOC sends too many Stash measurement commands, or if the Stash measurement command were to fail wihtin Caliptra for any reason, Caliptra should move to a Fatal Error state.

This will ensure that any SOC ROM/FW couldn't send a stash measurement that wasn't properly stored within Caliptra.

### PR DESCRIPTION
**Pull request by mhatrevi**
This change updates the logic when a stash measurement is requested beyond the max. supported. The new code raises a fatal error instead of continuing the boot sequence.

This change addresses issue# 1590.

### CODE REVIEW

**Comment by jhand2** [comment]
Left a comment on #1590 about this. This feels more like a behavior change than a bugfix.

**Review by jlmahowa-amd — DISMISSED** [review]

File: rom/dev/src/flow/cold_reset/fw_processor.rs [hunk 1]
```diff
@@ -271,10 +271,14 @@ impl FirmwareProcessor {
...
-                                "[fwproc] Maximum supported number of measurements already received, ignoring."
+                                "[fwproc] Maximum supported number of measurements already received."
                             );
                             txn.complete(false)?;
-                            continue;
+
+                            // Raise a fatal error on hitting the max. limit.
+                            // This ensures that any SOC ROM/FW couldn't send a stash measurement
+                            // that wasn't properly stored within Caliptra.
+                            return Err(CaliptraError::FW_PROC_MAILBOX_STASH_MEASUREMENT_MAX_LIMIT);
```
Comment (jhand2) [inline rom/dev/src/flow/cold_reset/fw_processor.rs:281]: We should make it very clear in the spec that this will happen

**Comment by nquarton** [comment]
Is it the case that this and the related, runtime behavior change are only going to affect 1.1 and beyond? Or are we backporting these to 1.0?

**Review by nquarton — DISMISSED** [review]

**Comment by mhatrevi** [comment]
> Is it the case that this and the related, runtime behavior change are only going to affect 1.1 and beyond? Or are we backporting these to 1.0?

Currently this is only for 1.1.

**Review by vsonims — APPROVED** [review]

**Pull request merged**